### PR TITLE
[browser] Add a special mode for WLAN captive portals. Contributes to JB#40777

### PR DESCRIPTION
**Pull request by NotKit**
This pull request adds support for starting browser with simplified UI and a separate profile enabled by -captiveportal flag, which is going to be used for opening Wi-Fi captive portal pages with separate browser profile to avoid leaking browser data.

### CODE REVIEW

**Review by rainemak — CHANGES_REQUESTED** [review]
Will push a pull request to your branch if that helps a bit with UI changes.

Please check other comments as well.

File: oneshot.d/browser-cleanup-startup-cache [hunk 1]
```diff
@@ -1,2 +1,3 @@
 #!/bin/sh
 rm -rf ~/.mozilla/mozembed/startupCache
+rm -rf /home/$USER/.mozilla/captiveportal/startupCache
```
Comment (okodron) [inline oneshot.d/browser-cleanup-startup-cache:3]: I think `/home/$USER` should be replaced with `~`.

Comment (rainemak) [reply oneshot.d/browser-cleanup-startup-cache:3]: Yes

File: src/main.cpp [hunk 2]
```diff
@@ -107,7 +110,10 @@ Q_DECL_EXPORT int main(int argc, char *argv[])
     //% "Browser"
     view->setTitle(qtTrId("sailfish-browser-ap-name"));
 
-    app->setApplicationName(QString("sailfish-browser"));
+    if (BrowserApp::captivePortal())
+        app->setApplicationName(QString("sailfish-browser-captiveportal"));
```
Comment (okodron) [inline src/main.cpp:114]: Maybe it's better to `QStringLiteral` macro

File: src/qtmozembed/declarativewebpage.cpp [hunk 3]
```diff
@@ -68,8 +69,10 @@ DeclarativeWebPage::DeclarativeWebPage(QObject *parent)
     addMessageListener(gLinkAddedMessage);
     addMessageListener(gFindMessage);
     addMessageListener(gContentOrientationChanged);
+    addMessageListener(gOpenLink);
 
     loadFrameScript("chrome://embedlite/content/embedhelper.js");
```
Comment (rainemak) [inline src/qtmozembed/declarativewebpage.cpp:74]: This only needed when captive portal browser app is active. I'd not load other wise.

Hence, maybe script should be called something like: "captiveportal.js"

File: src/qtmozembed/declarativewebpage.cpp [hunk 4]
```diff
@@ -68,8 +69,10 @@ DeclarativeWebPage::DeclarativeWebPage(QObject *parent)
     addMessageListener(gLinkAddedMessage);
     addMessageListener(gFindMessage);
     addMessageListener(gContentOrientationChanged);
```
Comment (rainemak) [inline src/qtmozembed/declarativewebpage.cpp:71]: All this message listener could be skipped when not captive portal.

File: src/pages/components/WebView.qml [hunk 5]
```diff
@@ -332,6 +335,11 @@ WebContainer {
                     } else {
                         webView.findInPageHasResult = false
                     }
+                    break
+                }
+                case "embed:OpenLink": {
```
Comment (rainemak) [inline src/pages/components/WebView.qml:340]: Just add comment here that this listener is only registered for the captive portal handler.

File: src/pages/webviewframescript.js [hunk 6]
```diff
@@ -0,0 +1,14 @@
...
+ */
+
+Components.utils.import("resource://gre/modules/Services.jsm");
+
+Services.scriptloader.loadSubScript("chrome://embedlite/content/ClickEventBlocker.js");
+
+let global = this;
+
+// This will send "embed:OpenLink" message when a link is clicked.
+ClickEventBlocker.init(global, {sameOrigin: true});
```
Comment (rainemak) [inline src/pages/webviewframescript.js:14]: allowNavigationInSameOrigin istead of "sameOrigin"

File: src/pages/components/CaptivePortalToolBar.qml [hunk 7]
```diff
@@ -0,0 +1,178 @@
...
+    Row {
+        id: toolsRow
+        width: parent.width
+        height: browserPage.isPortrait ? scaledPortraitHeight : scaledLandscapeHeight
+        leftPadding: Theme.paddingLarge
+
+        Browser.ExpandingButton {
+            id: backIcon
+            expandedWidth: toolBarRow.iconWidth
+            icon.source: "image://theme/icon-m-back"
```
Comment (rainemak) [inline src/pages/components/CaptivePortalToolBar.qml:101]: This has now two times Theme.horizontalPageMargin.

**Comment by rainemak** [comment]
Added small comments. Please rebase as well.

**Comment by rainemak** [comment]
Added review feedback in terms of changes here:
https://github.com/NotKit/sailfish-browser/pull/2

Feel free to squash that. Added "Sign in" label, doubled overlay height, removed "lockpad", removed some other dead code.

**Comment by rainemak** [comment]
And this has broken orientation handling.

**Comment by rainemak** [comment]
See also: https://git.sailfishos.org/mer-core/embedlite-components/merge_requests/81

File: src/BrowserWindow.qml [hunk 8]
```diff
@@ -0,0 +1,118 @@
...
+    _defaultPageOrientations: !largeScreen ? Orientation.Portrait : Orientation.LandscapeMask | Orientation.Portrait
+    _defaultLabelFormat: Text.PlainText
+    _clippingItem.opacity: 1.0
+    _resizeContent: !window.rootPage.active
+    _mainWindow: webView
+    _backgroundVisible: false
+    _opaque: false
+
+    cover: null
+    initialPage: Component {
```
Comment (rainemak) [inline src/BrowserWindow.qml:57]: This is extra here. Please remove it.

File: src/BrowserWindow.qml [hunk 9]
```diff
@@ -0,0 +1,118 @@
...
+            if (cover != null && window.webView) {
+              window.webView.clearSurface();
+            }
+            cover = null
+        }
+    }
+
+    allowedOrientations: defaultAllowedOrientations
+    // For non large screen fix cover to portrait.
+    _defaultPageOrientations: !largeScreen ? Orientation.Portrait : Orientation.LandscapeMask | Orientation.Portrait
```
Comment (rainemak) [inline src/BrowserWindow.qml:48]: Sorry my fault... this has dropped "!largeScreen && coverMode"

**Review by rainemak — APPROVED** [review]
Looking good now! Thank you for the first browser contribution!

**Pull request merged**